hack/netpol/pkg/utils/k8s_util.go

```diff
@@ -0,0 +1,270 @@
+	log.Infof("creating/updating deployment %s in ns %s", deploymentName, ns)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deploymentName,
+			Labels:    labels,
+			Namespace: ns,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{MatchLabels: labels},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:    labels,
+					Namespace: ns,
+				},
+				Spec: v1.PodSpec{
+					TerminationGracePeriodSeconds: &zero,
+					Containers: []v1.Container{
+						{
+							Name:            "c80",
+							Image:           "python:latest",
+							Command:         []string{"python", "-m", "http.server", "80"},
+							SecurityContext: &v1.SecurityContext{},
+							Ports: []v1.ContainerPort{
+								v1.ContainerPort{
+									ContainerPort: 80,
+									Name:          "serve-80",
+								},
+							},
+						},
+						{
+							Name:            "c81",
+							Image:           "python:latest",
+							Command:         []string{"python", "-m", "http.server", "81"},
+							SecurityContext: &v1.SecurityContext{},
+							Ports: []v1.ContainerPort{
+								v1.ContainerPort{
+									ContainerPort: 81,
+									Name:          "serve-81",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	d, err := k.ClientSet.AppsV1().Deployments(ns).Create(deployment)
+	if err == nil {
+		log.Infof("created deployment %s in namespace %s", d.Name, ns)
+		return d, nil
+	}
+
+	log.Debugf("unable to create deployment %s in ns %s, let's try update instead", deployment.Name, ns)
+	d, err = k.ClientSet.AppsV1().Deployments(ns).Update(d)
+	if err != nil {
+		log.Debugf("unable to update deployment %s in ns %s: %s", deployment.Name, ns, err)
+	}
+	return d, err
+}
+
+// CleanNetworkPolicies is a convenience function for deleting network policies before startup of any new test.
+func (k *Kubernetes) CleanNetworkPolicies(namespaces []string) {
+	for _, ns := range namespaces {
+		l, err := k.ClientSet.NetworkingV1().NetworkPolicies(ns).List(metav1.ListOptions{})
+		if err != nil {
+			log.Errorf("unable to list network policies in ns %s: %s", ns, err)
+			panic("failed at deleting netpols")
+		}
+		for _, np := range l.Items {
+			log.Infof("deleting network policy %s in ns %s", np.Name, ns)
+			err = k.ClientSet.NetworkingV1().NetworkPolicies(np.Namespace).Delete(np.Name, nil)
+			if err != nil {
+				log.Errorf("unable to delete network policy %s: %s", np.Name, err)
+				panic("deletion failed")
+			}
+		}
+	}
+}
+
+// CreateOrUpdateNetworkPolicy is a convenience function for upsdating/creating netpols.  Updating is important since
+// some tests update a network policy to confirm that mutation works with a CNI.
+func (k *Kubernetes) CreateOrUpdateNetworkPolicy(ns string, netpol *v1net.NetworkPolicy) (*v1net.NetworkPolicy, error) {
+	log.Infof("creating/updating network policy %s in ns %s", netpol.Name, ns)
+	netpol.ObjectMeta.Namespace = ns
```

Comment:
do we need to check if NS exists?

Reply:
yes but in reality this will be merged with the framework in K8s...